src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -213,12 +213,12 @@ void BaseCouplingScheme::initializeWithZeroInitialData(const DataMap &receiveDat
   }
 }
 
-PtrCouplingData BaseCouplingScheme::addCouplingData(const mesh::PtrData &data, mesh::PtrMesh mesh, bool requiresInitialization, bool communicateSubsteps)
+PtrCouplingData BaseCouplingScheme::addCouplingData(const mesh::PtrData &data, mesh::PtrMesh mesh, bool requiresInitialization, bool communicateSubsteps, CouplingData::Direction direction)
 {
   int             id = data->getID();
   PtrCouplingData ptrCplData;
   if (!utils::contained(id, _allData)) { // data is not used by this coupling scheme yet, create new CouplingData
-    ptrCplData = std::make_shared<CouplingData>(data, std::move(mesh), requiresInitialization, communicateSubsteps);
+    ptrCplData = std::make_shared<CouplingData>(data, std::move(mesh), requiresInitialization, communicateSubsteps, direction);
```

Comment:
I think we should here at least have some kind of assertion. Imagine there is already a `CouplingData` with `id = 3` in `_allData` for the `Direction::Send`. Then `if (!utils::contained(id, _allData)) ` would prevent adding a duplicate of this `CouplingData` with `id = 3` (good), even if the data is now in `Direction::Receive` (bad).